File: yongkeun/3week/진료순서정하기.js
function solution(emergency) {
  var answer = [];
  const oldEmergency = [...emergency]; //원본배열을 oldEmergency에 담아놓음 [...]을 쓰지 않으면 emergency를 바꾸면 oldEmergency도 바뀜. 그런 상황을 원하지 않으므로 [...]을 사용
  emergency.sort(function (a, b) {
    //내림차순정렬
    return b - a;
  });
  for (let i = 0; i < emergency.length; i++) {
    //돌면서
    answer.push(emergency.indexOf(oldEmergency[i]) + 1); //정렬된 배열의 값을 원본배열에서 찾아서 인덱스값을 출력하는데 몇 번째 순서인지 출력하므로 +1 함
  }
  return answer;
}

File: gang/week3/진료순서 정하기.java
import java.util.*;

class Solution {
    public int[] solution(int[] emergency) {
        int[] answer = new int[emergency.length];
        int[] sorted = Arrays.copyOf(emergency, emergency.length);  //emergency 배열을 복사해서 sorted배열 생성
        Arrays.sort(sorted);    //복사한 배열을 오름차순으로 정렬

        for (int i = 0; i < emergency.length; i++) {
            for (int j = 0; j < emergency.length; j++) {
                // 반복문으로 기존 emergency배열과 오름차순한 배열을 비교해서 같은 값이 나오면
                if (emergency[i] == sorted[j]) {
                    answer[i] = emergency.length - j;   //전체 길이에서 오름차순한 배열에서의 값 index를 빼서 기존 배열에 넣음
                    break;
                }
            }
        }
        return answer;
    }
}
